lib/fined.js

```diff
@@ -0,0 +1,141 @@
+const fs = require('fs');
+const path = require('path');
+
+const isArray = require('lodash.isarray');
+const isString = require('lodash.isstring');
+const isPlainObject = require('lodash.isplainobject');
+const isEmpty = require('lodash.isempty');
+const mapValues = require('lodash.mapvalues');
+const pick = require('lodash.pick');
+
+const expandTilde = require('expand-tilde');
+const parsePath = require('parse-filepath');
+
+
+function search(options, extensions, cwd) {
```

Comment:
The purpose of this function should be to find a single config file. The object iteration you've embedded here belongs outside of the `fined` library, or at the very least in a higher order method that calls the `search` primitive repeatedly.

That is to say you should not be able to do this:
```js
search({
  app: {
    path: path.resolve(cwd, 'test/fixtures/fined/app.js'),
    extension: '.js',
  },
  package: {
    path: path.resolve(cwd, 'package.json'),
    extension: '.json'
  }
}, ....);
```

This function should only take the value side of a single configFiles object.
```js
search({
  path: path.resolve(cwd, 'test/fixtures/fined/app.js'),
  extension: '.js',
}, ...)

search({
  path: path.resolve(cwd, 'package.json'),
  extension: '.json'
}, ...)
```